framework/db/BaseActiveRecord.php

```diff
@@ -236,10 +236,9 @@ public function optimisticLock()
      */
     public function __get($name)
     {
-        if (isset($this->_attributes[$name]) || array_key_exists($name, $this->_attributes)) {
-            return $this->_attributes[$name];
-        } elseif ($this->hasAttribute($name)) {
-            return null;
+        // getAttribute has better performance than hasAttribute for attributes that are set.
+        if ((($value = $this->getAttribute($name)) !== null) || $this->hasAttribute($name)) {
+            return $value;
```

Comment:
Is is okay, that `null` will be returned, when the left part of condition if `false`, and right - `true`?